pyteal/ast/router.py

```diff
@@ -475,7 +479,7 @@ def add_method_to_ast(
 
     def program_construction(self) -> Expr:
         if not self.conditions_n_branches:
-            raise TealInputError("ABIRouter: Cannot build program with an empty AST")
+            return Reject()
```

Comment:
I recommend adding commentary to `build_program()` and/or `compile_program()` which points out that if there is no AST or alternately if no methods or bare calls have been added, a trivial rejecting program will be constructed.